src/bridges/tokemak/DepositBridge.sol

```diff
@@ -0,0 +1,337 @@
+
+    //Check if the pool is EthPool because withdrawal function is different
+    if(asset == WETH){
+      tToken.withdraw(inputValue, false);
+    }else{
+      tToken.withdraw(inputValue);
+    }
+
+    // Asset balance after withdrawal for calculating outputValue
+    uint256 afterBalance = assetToken.balanceOf(address(this));
+
+    outputValue = afterBalance - beforeBalance;
+
+    //Approve Rollup Processor to withdraw asset
+    assetToken.approve(rollupProcessor, outputValue);
+
+    delete pendingInteractions[nonce];
+    for(uint i = 0;i < nonces.length;i++){
+      if(nonces[i] == nonce){
+        delete nonces[i];
+        break;
+      }
+    }
+  }
+
+  function _deposit(address tAsset, uint256 inputValue, address asset) private returns (uint256 outputValue,bool isAsync) {
+    // Approve asset to deposit in Tokemak Pool
+    IERC20(asset).approve(tAsset, inputValue);
+
+    Ttoken tToken = Ttoken(tAsset);
+
+    // Asset balance before withdrawal for calculating outputValue
+    uint256 beforeBalance = tToken.balanceOf(address(this));
+    
+    //Deposit in Tokemak Pool
+    tToken.deposit(inputValue);
+
+    // Asset balance after withdrawal for calculating outputValue
+    uint256 afterBalance = tToken.balanceOf(address(this));
+
+    // Output Value
+    outputValue = afterBalance - beforeBalance;
+
+    //Approve Rollup Processor to withdraw tAsset
+    tToken.approve(rollupProcessor, outputValue);
+  }
+  
+  function _isWithdrawal(address token) internal view returns(bool){
+    if(token == tWETH || token == tUSDC || token == tDAI || token == tFRAX || token == tWUST){
+      return true;
+    }
+    return false;
+  }
+  //Get asset address from tAsset address
+  function _getAssetAddress(address tToken) internal view returns (address){
+    if(tToken == tWETH){
+      return WETH;
+    }
+    if(tToken == tUSDC){
+      return USDC;
+    }
+    if(tToken == tFRAX){
+      return FRAX;
+    }
+    if(tToken == tDAI){
+      return DAI;
+    }
+    if(tToken == tWUST){
+      return wUST;
+    }
+    return address(0);
+  }
+  //Get tAsset address from asset address
+  function _getPoolAddress(address token) internal view returns (address){
+    if(token == WETH){
+      return tWETH;
+    }
+    if(token == USDC){
+      return tUSDC;
+    }
+    if(token == FRAX){
+      return tFRAX;
+    }
+    if(token == DAI){
+      return tDAI;
+    }
+    if(token == wUST){
+      return tWUST;
+    }
+    return address(0);
+  }
+
+  function finalise(
+    AztecTypes.AztecAsset calldata inputAssetA,
+    AztecTypes.AztecAsset calldata inputAssetB,
+    AztecTypes.AztecAsset calldata outputAssetA,
+    AztecTypes.AztecAsset calldata outputAssetB,
+    uint256 interactionNonce,
+    uint64 auxData
+  ) external payable override returns (uint256 outputValueA, uint256 outputValueB, bool interactionCompleted ) {
+    require(inputAssetA.assetType == AztecTypes.AztecAssetType.ERC20, "Tokemak DepositBridge: INVALID_ASSET_TYPE");
```

Comment:
Can you validate all of the input arguments in the same way as the convert function?